File: src/main/java/org/oewntk/yaml/in/VerbTemplateProcessor.java
/*
 * Copyright (c) $originalComment.match("Copyright \(c\) (\d+)", 1, "-")2021. Bernard Bou.
 */

package org.oewntk.yaml.in;

import org.oewntk.model.VerbTemplate;

import java.io.File;
import java.util.Map.Entry;

public class VerbTemplateProcessor extends YamProcessor1<VerbTemplate, Integer, String>
{
	private static final boolean DUMP = false;

	/**
	 * Constructor
	 *
	 * @param dir dir containing YAML files
	 */
	public VerbTemplateProcessor(final File dir)
	{
		super(dir);
		this.dir = dir;
	}

	@Override
	protected File[] getFiles()
	{
		return dir.listFiles((f) -> f.getName().matches("verbTemplates.yaml"));
	}

	@Override
	protected VerbTemplate processEntry(final String source, final Entry<Integer, String> entry)
	{
		Integer id = entry.getKey();
		String v = entry.getValue();
		if (DUMP)
		{
			Tracing.psInfo.println(id);
		}
		return new VerbTemplate(id, v);
	}
}

File: src/main/java/org/oewntk/yaml/in/VerbTemplateProcessor.kt
/*
 * Copyright (c) 2024. Bernard Bou.
 */
package org.oewntk.yaml.`in`

import org.oewntk.model.VerbTemplate
import org.oewntk.model.VerbTemplateId
import java.io.File

/**
 * Verb Templates processor
 *
 * @property dir dir containing YAML files
 */
class VerbTemplateProcessor(dir: File) : YamProcessor1<VerbTemplate, Int, String>(dir) {

    init {
        this.dir = dir
    }

    override val files: Array<File>
        get() = dir.listFiles { f -> f.name.matches("verbTemplates.yaml".toRegex()) }!!

    override fun processEntry(source: String?, entry: Pair<VerbTemplateId, String>): VerbTemplate {
        val id = entry.first
        val v = entry.second
        if (DUMP) {
            Tracing.psInfo.println(id)
        }
        return VerbTemplate(id, v)
    }

    companion object {

        private const val DUMP = false
    }
}

